packages/wallet/frontend/src/lib/hooks/useTransactions.ts

```diff
@@ -20,9 +31,16 @@ const defaultState = {
 }
 
 export const useTransactions = () => {
-  const router = useRouter()
-  const { accountId, walletAddressId, type, status, page } =
-    router.query as TransactionsQueryParams
+  const router = useTypedRouter(transactionListQuerySchema)
```

Comment:
We need to improve the `useTypedRouter` hook. Will have a look at it and get back.

Reply:
https://github.com/interledger/testnet/blob/1f43d89fb50074b56a2552e9350f80d30968af55/packages/wallet/frontend/src/lib/hooks/useTypedRouter.ts#L5

```ts
import { type NextRouter, useRouter } from 'next/router'
import { z } from 'zod'

export function useTypedRouter<T extends z.ZodSchema>(
  schema: T
): NextRouter & { query: z.infer<T> } {
  const { query, ...router } = useRouter()

  return {
    query: schema.parse(query) as z.infer<typeof schema>,
    ...router
  }
}
```

This way we don't need to use type assertions when using the hook. The `list` function arguments need to be updated as well.

Zod's `parse` function will throw an error when validating, so we need to take this into consideration as well. Once solution is that we ignore invalid items and only return the valid ones, without throwing an error. For example, if `accountId` is not a `uuid`, we do not return it.